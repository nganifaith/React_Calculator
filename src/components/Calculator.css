.App {
  background-color: #fff;
  width: 700px;
  border-radius: 16px;
  margin-top: 15px;
}

.row {
  display: flex;
}
