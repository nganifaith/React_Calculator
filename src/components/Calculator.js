import React, { useState } from 'react';
import './Calculator.css';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import Calculate from '../Logic/calculate';

const Calculator = () => {
  const [data, setData] = useState({
    next: null,
    total: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setData(Calculate(data, buttonName));
  };
  return (
    <div className="App">
      <h4>Lets do some math!</h4>
      <div className="calc">
        <Display result={data.next || data.total} />
        <ButtonPanel handleClick={handleClick} />
      </div>
    </div>
  );
};

export default Calculator;
