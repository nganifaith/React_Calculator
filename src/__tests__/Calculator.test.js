import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('renders pages correctly', () => {
  const calculator = renderer.create(<Calculator />).toJSON();
  expect(calculator).toMatchSnapshot();
});
