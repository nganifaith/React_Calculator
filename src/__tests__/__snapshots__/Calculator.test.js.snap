// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders pages correctly 1`] = `
<div
  className="App"
>
  <h4>
    Lets do some math!
  </h4>
  <div
    className="calc"
  >
    <div
      className="input-wrapper"
    >
      <div
        className="result"
      >
        <h3 />
      </div>
    </div>
    <div>
      <div
        className="row"
      >
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          AC
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          +/-
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          %
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f5923e",
            }
          }
          type="button"
        >
          /
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          7
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          8
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          9
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f5923e",
            }
          }
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          4
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          5
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          6
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f5923e",
            }
          }
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          1
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          2
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          3
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f5923e",
            }
          }
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="button-wrapper wide"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          0
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
            }
          }
          type="button"
        >
          .
        </button>
        <button
          className="button-wrapper "
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#f5923e",
            }
          }
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
